src/monster.cpp

```diff
@@ -1503,6 +1503,11 @@ bool monster::melee_attack( Creature &target, float accuracy )
         return false;
     }
 
+    if (!sees(target) && posz() != target.posz()) {
+        debugmsg(string_format("Z-Level view violation: %s tried to attack %s.", disp_name(),
+            target.disp_name()));
+    }
+
```

Comment:
Tiny micro-optimization, maybe faster?
debugmsg will do the string formatting for you, but this may need further styling. 
```suggestion
    if( posz() != target.posz() && !sees( target ) ) {
        debugmsg( "Z-Level view violation: %s tried to attack %s.", disp_name(), target.disp_name() );
    }

```